ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageManager.java

```diff
@@ -67,8 +65,8 @@ public boolean shutdownBlobStore(PartitionId id) {
 
   @Override
   public Store getStore(PartitionId id) {
-    Store store = partitionToStore.get(id);
-    return (store != null && store.isStarted()) ? store : null;
+    createAndStartBlobStoreIfAbsent(id);
```

Comment:
As discussed offline, let's consider the calls to shutdownBlobStore and removeStore in VCR replication

Reply:
As a fix i have made the CloudBlobStore operations thread safe. Both removestore and shutdownstore are there as is in the add/remove replica flow. But a getstore will return null, if the store is not started.